File: 2697-lexicographically-smallest-palindrome/2697-lexicographically-smallest-palindrome.cpp
class Solution {
public:
    string makeSmallestPalindrome(string s) {
        int i=0,j=s.size()-1;
        while(i<j){
            if(s[i] != s[j]){
                char x = min(s[i],s[j]);
                s[i] = s[j] = x;
            }
            i++;
            j--;
        }
        return s;
    }
};

File: 2697-lexicographically-smallest-palindrome/2697-lexicographically-smallest-palindrome.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        temp = list(s)
        i = 0
        j = len(temp) - 1
        
        while i < j:
            if temp[i] != temp[j]:
                temp[i] = temp[j] = min(temp[i], temp[j])
            i += 1
            j -= 1
            
        return "".join(temp)
